api/v1alpha1/hosted_controlplane.go

```diff
@@ -90,6 +90,10 @@ type HostedControlPlaneStatus struct {
 	// +kubebuilder:validation:Optional
 	Version string `json:"version,omitempty"`
 
+	// KubeConfig is a reference to the secret containing the default kubeconfig
+	// for this control plane.
+	KubeConfig *SecretReference `json:"kubeConfig,omitempty"`
```

Comment:
why don't we use `corev1.LocalObjectReference` consistently with the other secret refs we do?

Reply:
Spoke about this and agree, fixed